data/src/main/java/com/linkedin/data/schema/PdlBuilder.java

```diff
@@ -149,7 +149,7 @@ PdlBuilder writeProperties(List<String> prefix, Map<String, Object> properties)
       {
         // Favor @x.y.z = "value" property encoding style over @x = { "y": { "z": "value" } }
```

Comment:
Most of the comments in this method are now invalid. You should update them.

> Favor @x.y.z = "value" property encoding style over @<!-- -->x = { "y": { "z": "value" } }
> encode non-empty value property like @x.y.z = "value"
> encode empty value property like @x.y = {}

I'd also suggest cleaning this up.

> Use shorthand for boolean true.  Instead of writing "@<!-- -->deprecated = true",
> write "@<!-- -->deprecated".

Also, remove this from the method javadoc.

> TODO: Support configuration for preferring collapsed "map" style over expanded "path" style.

Reply:
Your changes still don't address most of this.